captcha.py

```
@@ -0,0 +1,241 @@
+
+# ── Challenge generation ──────────────────────────────────────
+
+
+def generate_captcha() -> tuple[str, str]:
+    """Create a lens-style CAPTCHA challenge.
+
+    Returns (answer, token).
+      - answer: 7-char string (for rendering the image)
+      - token:  HMAC commitment to the correct answer
+    """
+    answer = "".join(random.choices(CHAR_POOL, k=CAPTCHA_LEN))
```

Ruff (0.15.9)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)